sources/packages/backend/apps/api/src/route-controllers/designation-agreement/models/designation-agreement.dto.ts

```diff
@@ -12,38 +15,117 @@ import {
   ValidateIf,
   ValidateNested,
 } from "class-validator";
-import { UpdateDesignationLocation } from "./designation-agreement.model";
 
 /**
  * Approve/Deny a designation agreement.
  * startDate, endDate and locationsDesignations used only for approval.
  */
-export class UpdateDesignationAPIInDto {
+export class UpdateDesignationAPIInDTO {
   @IsEnum(DesignationAgreementStatus)
   designationStatus: DesignationAgreementStatus;
   @ValidateIf(
-    (value: UpdateDesignationAPIInDto) =>
+    (value: UpdateDesignationAPIInDTO) =>
       value.designationStatus === DesignationAgreementStatus.Approved,
   )
   @IsNotEmpty()
   @IsDateString()
   startDate?: string;
   @ValidateIf(
-    (value: UpdateDesignationAPIInDto) =>
+    (value: UpdateDesignationAPIInDTO) =>
       value.designationStatus === DesignationAgreementStatus.Approved,
   )
   @IsNotEmpty()
   @IsDateString()
   endDate?: string;
   @ValidateIf(
-    (value: UpdateDesignationAPIInDto) =>
+    (value: UpdateDesignationAPIInDTO) =>
       value.designationStatus === DesignationAgreementStatus.Approved,
   )
   @IsNotEmpty()
   @IsArray()
   @ValidateNested({ each: true })
-  @Type(() => UpdateDesignationLocation)
-  locationsDesignations?: UpdateDesignationLocation[];
+  @Type(() => DesignationLocationAPIInDTO)
+  locationsDesignations?: DesignationLocationAPIInDTO[];
+  @IsNotEmpty()
+  @MaxLength(NOTE_DESCRIPTION_MAX_LENGTH)
+  note: string;
+}
+
+/**
+ * This DTO contains dynamic data that must
+ * be validated by the form.io dryrun validation.
+ */
+export class SubmitDesignationAgreementAPIInDTO {
```

Comment:
For all the *APIInDTO the it's properties need class-validator decorators at the minimum `@Allow()` otherwise the API will read the property value as null.